cdap-security/src/main/java/co/cask/cdap/security/authz/AuthorizationPlugin.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authz;
+
+import co.cask.cdap.proto.id.EntityId;
+
+import java.util.Set;
+
+/**
+ * Authorization plugin interface.
+ */
+public interface AuthorizationPlugin {
+  /**
+   * Checks if a user is allowed to perform a set of actions on an entity.
+   *
+   * @param entity the entity
+   * @param user the user
+   * @param actions the actions (wildcard: '*')
+   * @return true if the user is allowed to perform all of the actions on the entity
+   */
+  boolean authorized(EntityId entity, String user, Set<Action> actions);
```

Comment:
If we want to check if the user has a READ access on a specific entity, we need to construct a Set and add the action to it and call the ``authorized`` method. Should we provide a way to call ``authorized`` for a single ``Action`` in addition to ``Set<Action>``. 

My second comment was about the need for a method that takes in ``Set<Action>`` in the first place. I guess this is required since there is no concrete hierarchy in Actions. For example, I guess we CAN'T order the Actions in this way - All > Write > Read. So if we want to check if a user has read and write access, we need to make a call providing both the actions to the authorized method.